packages/react/src/primitives/SearchField/useSearchField.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+
+import { isFunction } from '../shared/utils';
+import { SearchFieldProps } from '../types';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
+
+const ESCAPE_KEY = 'Escape';
+const ENTER_KEY = 'Enter';
+const DEFAULT_KEYS = [ESCAPE_KEY, ENTER_KEY];
+
+export const useSearchField = ({
+  onSubmit,
+  onClear,
+  externalRef,
+}: Partial<
+  SearchFieldProps & { externalRef: React.ForwardedRef<HTMLInputElement> }
+>) => {
+  const [value, setValue] = React.useState<string>('');
+  const internalRef = React.useRef<HTMLInputElement>(null);
+  const composeRefs = useComposeRefsCallback({ externalRef, internalRef });
+
+  const onClearHandler = React.useCallback(() => {
+    setValue('');
+    internalRef?.current?.focus();
```

Comment:
I'd prefer to be safe and check if `internalRef.current` is not undefined

```suggestion
    if (internalRef && internalRef.current) {
      internalRef.current.focus()
    }
```

Reply:
I think the optional chaining should be sufficient here